src/AtelierTomato.Markov.Storage.Sqlite/SqliteSentenceAccess.cs

```diff
@@ -39,13 +49,21 @@ await connection.ExecuteAsync($@"
 
 		public async Task<IEnumerable<Sentence>> ReadNextRandomSentences(int amount, List<string> prevList, List<IObjectOID> previousIDs, SentenceFilter filter, string? keyword = null)
 		{
+			var sbOIDs = GenerateOIDStringBuilder(filter);
+
+			IEnumerable<string>? authors = null;
+			if (filter.Authors is not null && filter.Authors.Any())
+			{
+				authors = filter.Authors.Select(a => a.ToString());
+			}
+
 			await using var connection = new SqliteConnection(options.ConnectionString);
 			connection.Open();
 
 			var result = await connection.QueryAsync<SentenceRaw>($@"
 SELECT {nameof(Sentence.OID)}, {nameof(Sentence.Author)}, {nameof(Sentence.Date)}, {nameof(Sentence.Text)} FROM {nameof(Sentence)} WHERE
-( @oid IS NULL OR {nameof(Sentence.OID)} LIKE @oid || '%' ) AND
-( @author IS NULL OR {nameof(Sentence.Author)} LIKE @author || '%' ) AND
+{sbOIDs}

```

Comment:
this is very very close to being an SQL injection vulnerability, this is very very bad.

Reply:
next time i'm killing myself on the spot